### PR TITLE
Add New Relic to Footlocker & H&R Block apps.

### PR DESCRIPTION
**Pull request by DFurnes**
This pull request adds optional New Relic support to the Longshot application module. If the `with_newrelic` argument is set, a New Relic Heroku add-on & appropriate environment variables will be set for that application.

References #49.

### CODE REVIEW

**Comment by DFurnes** [comment]
Going to test this out on H&R Block first in 9eea907:

```
+ module.longshot.module.hrblock.heroku_addon.newrelic
    id:              "<computed>"
    app:             "longshot-hrblock"
    config_vars.#:   "<computed>"
    name:            "<computed>"
    plan:            "newrelic:wayne"
    provider_id:     "<computed>"

~ module.longshot.module.hrblock.heroku_app.app
    config_vars.0.%:                       "24" => "27"
    config_vars.0.NEW_RELIC_APP_NAME:      "" => "longshot-hrblock"
    config_vars.0.NEW_RELIC_LICENSE_KEY:   "" => "******** SECRET ********"
    config_vars.0.NEW_RELIC_LOG:           "" => "stdout"

~ module.longshot.module.longshot-footlocker-internal.heroku_app.app
    config_vars.0.%:                       "24" => "27"

~ module.longshot.module.longshot-footlocker.heroku_app.app
    config_vars.0.%:                       "24" => "27"

~ module.longshot.module.longshot-qa.heroku_app.app
    config_vars.0.%:                       "24" => "27"
```

File: longshot/application/main.tf [hunk 1]
```diff
@@ -81,6 +84,11 @@ variable "database_size_gb" {
   default     = 100
 }
 
+variable "with_newrelic" {
```
Comment (sheyd) [inline longshot/application/main.tf]: Thought from something in you mentioned in another PR related to this issue the other day. Is this module maybe better put in `shared` module, if we want to use it across properties? Or are we not sure we want to stick with New Relic, and this is our first test to see how we like it on Heroku?

Comment (DFurnes) [reply longshot/application/main.tf]: Sorry, which module are you referring to here? The overall `longshot/application` module?

Comment (DFurnes) [reply longshot/application/main.tf]: Chatted this through IRL. For anyone else who's following along – the `with_newrelic` variable here is just an argument for the module (think of it as a parameter you'd pass a function in other languages), so it's inherently tied to this module and isn't something we could reasonably extract.

We could definitely extract pieces of this `longshot/application` module into more reusable chunks for elsewhere (so say, bundling the `aws_ssm_parameter` & `heroku_addon` into a `shared/heroku-newrelic` module that exports the expected config values) and I'm planning on tackling that further in #28 after this app is wrapped up.

**Comment by DFurnes** [comment]
Hmm, frustratingly the New Relic add-on doesn't install the agent until you complete a build (confirmed by running `php -i` on longshot-hrblock). I think this hasn't bitten us before since we have the New Relic agent on QA instances where we create Heroku builds, but does pose problem here if we want to avoid installing (and paying for) New Relic on QA here.

One solution is to add `ext-newrelic` to this application's `composer.json` (and that actually may be closer to what we want since it doesn't auto-provision a weird parallel Heroku New Relic account for each app), or we could always include the add-on but only set our "real" account on prod...

**Comment by DFurnes** [comment]
I'm leaning towards using the New Relic extension (and adding it to our Homestead's via an `after.sh` step) since that doesn't create the weird parallel accounts & clobber environment variables.

**Comment by sheyd** [comment]
This Approach Approved by 
download

**Comment by sheyd** [comment]
YAAAAS!

**Pull request merged**